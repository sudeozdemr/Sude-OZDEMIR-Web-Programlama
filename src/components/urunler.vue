<template>
  <div>
    <div class="container" v-if="products && products.length">
      <router-link
        :to="{ name: 'urun', params: { id: product.id } }"
        v-for="product in products"
        :key="product.id"
      >
        <div class="box">
          <div class="imgDiv">
            <img :src="product.image" alt="" />
          </div>
          <h3 class="urunHeader">{{ product.urunAdi }}</h3>
          <p>
            {{ product.urunAciklama }}
          </p>
          <div class="urunFiyatDiv">
            <span
              ><b>{{ Number.parseFloat(product.fiyat).toFixed(2) }} TL</b></span
            >
          </div>
          <div class="sepeteGitIcon">
            <span
              ><FontAwesomeIcon icon="shopping-cart"></FontAwesomeIcon
            ></span>
          </div>
          <div class="ucretsizKargo">
            <span><b>ÜCRETSİZ KARGO</b></span>
          </div>
        </div>
      </router-link>
    </div>
    <div v-else class="container">
      <img class="box"
        v-for="i in 9"
        :key="i"
        src="https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/source.gif"
        alt="https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/source.gif"
      />
    </div>
  </div>
</template>

<script>
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import { library } from "@fortawesome/fontawesome-svg-core";
import { fas } from "@fortawesome/free-solid-svg-icons";
library.add(fas);
import { mapActions } from "vuex";
export default {
  components: {
    FontAwesomeIcon,
  },
  data() {
    return {
      products: [],
    };
  },
  created() {
    this.fetchProducts().then((data) => {
      this.products = data;
    });
  },
  methods: {
    ...mapActions(["fetchProducts"]),
  },
};
</script>
<style scoped>
.container {
  margin: 0 25px 0 25%;
  width: 70%;
  background: #e8e8e8;
}
.box {
  display: inline-block;
  width: 270px;
  /*border: 1px solid gray;*/
  position: relative;
  margin: 15px;
  background: white;
}
.imgDiv {
  width: 100%;
}
.imgDiv img {
  width: 100%;
}
.urunHeader {
  text-align: center;
  color: #14569b;
}
.urunHeader:hover {
  text-decoration: underline;
  cursor: pointer;
}
.box p {
  text-align: center;
  padding-bottom: 45px;
  color: #7389a3;
}
.urunFiyatDiv {
  position: absolute;
  bottom: 0;
  left: 10px;
  color: #444448;
  font-size: 20px;
  margin: 15px;
}
.sepeteGitIcon {
  position: absolute;
  bottom: 0;
  right: 15px;
  color: #ff7920;
  margin: 0 5px 15px 0;
}
.ucretsizKargo {
  position: absolute;
  top: 0;
  right: -5px;
  width: 70px;
  font-size: 12px;
  height: auto;
  background: #1bbc9c;
  color: white;
  text-align: center;
  padding: 7px 0;
}
.ucretsizKargo::after {
  content: "";
  position: absolute;
  display: block;
  top: 100%;
  right: 0;
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 0 5px 5px 0;
  border-color: transparent #6c6c7c transparent transparent;
}
.yeniUrun {
  position: absolute;
  top: 10px;
  left: -5px;
  width: 70px;
  font-size: 12px;
  padding: 7px 0;
  box-shadow: 3px 3px #888888;
  background: #ff7920;
  color: white;
  text-align: center;
}
.yeniUrun::before {
  content: "";
  position: absolute;
  display: block;
  top: 100%;
  left: 0;
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 0 5px 5px 0;
  border-color: transparent #6c6c7c transparent transparent;
}
</style>