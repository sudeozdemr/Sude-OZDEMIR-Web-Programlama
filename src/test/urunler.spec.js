import { mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex"
import urunler from "../components/urunler.vue";
describe("Ürünler Listesi", () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    const mockFunction = jest.fn();
    const products = [{ "id": 1, "urunAdi": "ali", "urunAciklama": "Saydam Kontakt Lens Bilgisayar karşısında ve klimalı ortamlarda benzersiz üstün konfor...", "fiyat": 119.90, "image": "https://cdn.lensmarket.com/ext/urun/oasys-1-l.jpg" },
        { "id": 2, "urunAdi": "Acuvue OASYS ® 1-Day 30 lu Kutu", "urunAciklama": "Günlük Kontakt Lens ACUVUE OASYS 1-Day silikon hidrojel günlük kontakt lensler en yüksek...", "fiyat": 149.90, "image": "https://cdn.lensmarket.com/ext/urun/oasys-1day-1-l.jpg" },
        { "id": 3, "urunAdi": "Acuvue OASYS ® with Transitions™", "urunAciklama": "Türünün ilk örneği, ışığa duyarlı lens. Time'a Göre 2018'in en iyi buluşlarından biri!", "fiyat": 41.90, "image": "https://cdn.lensmarket.com/ext/urun/oasys-transitions-1-l.jpg" },
    ];
    const fetchProducts = () => {
        return new Promise(resolve => {
            resolve(products)
        })
    };
    const wrapper = mount(urunler, {
        stubs: ["router-link"],
        localVue,
        store: new Vuex.Store({
            actions: {
                fetchProducts: fetchProducts
            },
        })
    })

    describe("Liste Kontrolü", () => {
        it("Ürün 1'in adı : ali :", () => {
            expect(wrapper.text()).toContain("ali");
        });
        it("Ürün 2'nin açıklaması : Günlük Kontakt Lens ACUVUE OASYS 1-Day silikon hidrojel günlük kontakt lensler en yüksek... :", () => {
            expect(wrapper.text()).toContain("Günlük Kontakt Lens ACUVUE OASYS 1-Day silikon hidrojel günlük kontakt lensler en yüksek...");
        });

        it("Ürün 3'ün fiyatı : 41.90", () => {
            expect(wrapper.text()).toContain("41.90");
        });
        test("Snapshot Testi", () => {
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
    test('fetchProducts Fonsiyonu Ürün listesini çözümler', () => {
        return expect(fetchProducts()).resolves.toBe(products);
    });
})