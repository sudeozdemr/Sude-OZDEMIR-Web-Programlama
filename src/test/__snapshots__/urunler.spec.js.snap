// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ürünler Listesi Liste Kontrolü Snapshot Testi 1`] = `
"<div>
  <div class=\\"container\\">
    <router-link-stub to=\\"[object Object]\\">
      <div class=\\"box\\">
        <div class=\\"imgDiv\\"><img src=\\"https://cdn.lensmarket.com/ext/urun/oasys-1-l.jpg\\" alt=\\"\\"></div>
        <h3 class=\\"urunHeader\\">ali</h3>
        <p>
          Saydam Kontakt Lens Bilgisayar karşısında ve klimalı ortamlarda benzersiz üstün konfor...
        </p>
        <div class=\\"urunFiyatDiv\\"><span><b>119.90 TL</b></span></div>
        <div class=\\"sepeteGitIcon\\"><span><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"shopping-cart\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 576 512\\" class=\\"svg-inline--fa fa-shopping-cart fa-w-18\\"><path fill=\\"currentColor\\" d=\\"M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z\\" class=\\"\\"></path></svg></span></div>
        <div class=\\"ucretsizKargo\\"><span><b>ÜCRETSİZ KARGO</b></span></div>
      </div>
    </router-link-stub>
    <router-link-stub to=\\"[object Object]\\">
      <div class=\\"box\\">
        <div class=\\"imgDiv\\"><img src=\\"https://cdn.lensmarket.com/ext/urun/oasys-1day-1-l.jpg\\" alt=\\"\\"></div>
        <h3 class=\\"urunHeader\\">Acuvue OASYS ® 1-Day 30 lu Kutu</h3>
        <p>
          Günlük Kontakt Lens ACUVUE OASYS 1-Day silikon hidrojel günlük kontakt lensler en yüksek...
        </p>
        <div class=\\"urunFiyatDiv\\"><span><b>149.90 TL</b></span></div>
        <div class=\\"sepeteGitIcon\\"><span><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"shopping-cart\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 576 512\\" class=\\"svg-inline--fa fa-shopping-cart fa-w-18\\"><path fill=\\"currentColor\\" d=\\"M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z\\" class=\\"\\"></path></svg></span></div>
        <div class=\\"ucretsizKargo\\"><span><b>ÜCRETSİZ KARGO</b></span></div>
      </div>
    </router-link-stub>
    <router-link-stub to=\\"[object Object]\\">
      <div class=\\"box\\">
        <div class=\\"imgDiv\\"><img src=\\"https://cdn.lensmarket.com/ext/urun/oasys-transitions-1-l.jpg\\" alt=\\"\\"></div>
        <h3 class=\\"urunHeader\\">Acuvue OASYS ® with Transitions™</h3>
        <p>
          Türünün ilk örneği, ışığa duyarlı lens. Time'a Göre 2018'in en iyi buluşlarından biri!
        </p>
        <div class=\\"urunFiyatDiv\\"><span><b>41.90 TL</b></span></div>
        <div class=\\"sepeteGitIcon\\"><span><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"shopping-cart\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 576 512\\" class=\\"svg-inline--fa fa-shopping-cart fa-w-18\\"><path fill=\\"currentColor\\" d=\\"M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z\\" class=\\"\\"></path></svg></span></div>
        <div class=\\"ucretsizKargo\\"><span><b>ÜCRETSİZ KARGO</b></span></div>
      </div>
    </router-link-stub>
  </div>
</div>"
`;
